### PR TITLE
Flake8 fixes and black autoformatting

### LINKED ISSUE #28: Undefined names in `verydeepvae.data_tools.data_transformations_2d` module
The `verydeepvae.data_tools.data_transformations_2d` module contains this snippet of code

https://github.com/high-dimensional/3d_very_deep_vae/blob/b8c4763a12a8da16885c16245e9591fc37d59734/verydeepvae/data_tools/data_transformations_2d.py#L54-L61

in which the nams `SimulateMissingnessAugmentation`, `LoadJSONd` and `SimulateJSONMissingnessAugmentation` are all not previously defined. They also all do not seem to be defined anywhere else in the codebase.

The single `create_data_transformations_2d` function in this module is not currently called anywhere - @r-gray should this just be removed?

### PR DESCRIPTION
**Pull request by matt-graham**
Autoformats whole codebase using [black](https://github.com/psf/black) to standardize styling and ensure more meaningful output from `flake8` checks by reducing styling related warnings / violations. Some manual fixes to problems identified by running `flake8` are also included:

  * Missing imports of `Any` from `typing` in a few places
  * Removing definitions of unused variables in a few places
  * Missing import of `progress_bar` from `verydeepvae.misc.visuals` in `verydeepvae/data_tools/data_handling.py`
  * Missing import of `ImageOps` from `PIL` in `verydeepvae/data_tools/jpeg_dataset.py`
  * Replacing use of `quit` with exception based approach in `verydeepvae/data_tools/jpeg_dataset.py`
  * Using more descriptive variable name `label` in place of ambiguous single character name `l` in `verydeepvae/data_tools/jpeg_dataset.py`
  * Removing unused aliased import of `torch.nn.functional` in `in `verydeepvae/graph_components/pooling_block.py`
  * Removing overwritten import of `DataLoader` from `monai.data` in `verydeepvae/orchestration/training_script_vae_new.py`
  * Moving `mpl.use("AGG")` line in `verydeepvae/misc/visuals.py` to ensure all import statements come first

The module `verydeepvae/data_tools/data_transformations_2d.py` the only function in which does not appear to be used anywhere and includes various undefined names is also removed which fixes #28.

The `tox` `check` environment (which is run in the GitHub Actions continuous integration) is updated to also run `flake8`. Currently this checks fails due to the undefined name issue described in #29 which still needs fixing.

The commit has corresponding the black autoformatting is [added to a new `.git-blame-ignore-revs` file to avoid polluting the blame with these changes](https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view). This means this pull-request will need to be merged without squashing / rebasing so the commit hash included in the file remains correct.

### CODE REVIEW

**Comment by matt-graham** [comment]
@r-gray any chance you could this a quick check over? Ideally would like to merge this in today as the changes to remove configuration options you highlighted as deprecated in the [wiki page](https://github.com/high-dimensional/3d_very_deep_vae/wiki/Configuration-file-options) touch a lot of the same code so ideally PR will be on a branch after this is merged into main to avoid lots of merge conflicts. [This diff page](https://github.com/high-dimensional/3d_very_deep_vae/pull/34/files/c244f7c1350e6d54bb11cfb36c619d4683fdc991..f72697c1e1b2187fe4f8df1e9f24d90ae2f886db) shows only the commits which don't correspond to running the `black` autoformatter which are the more relevant from a review perspective (as the autoformatting shouldn't have any effect on code behaviour).

**Comment by high-dimensional-admin** [comment]
I'm looking now!

**Comment by high-dimensional-admin** [comment]
That looks absolutely fine, please go ahead!

**Comment by r-gray** [comment]
Sorry, and again from my account... This all looks fine, please go ahead!

**Comment by matt-graham** [comment]
> Sorry, and again from my account... This all looks fine, please go ahead!

There should be a series of check boxes like the following



as I've requested a review from you - if you select the 'Approve' option and submit review that should flag the PR as ready to merge 🙂
